keys/keys.go

```diff
@@ -261,10 +261,10 @@ func ValidateRangeMetaKey(key proto.Key) error {
 	prefix, body := proto.Key(key[:len(Meta1Prefix)]), proto.Key(key[len(Meta1Prefix):])
 
 	if prefix.Equal(Meta2Prefix) {
-		if body.Less(proto.KeyMax) {
-			return nil
+		if proto.KeyMax.Less(body) {
+			return NewInvalidRangeMetaKeyError("body of meta2 range lookup is > KeyMax", key)
```

Comment:
code above: `if KeyMax < key { error }`, so `> KeyMax` errors.
old code: `if key < KeyMax { no error } else { error }` (so everything `>=KeyMax` errors).

This isn't just refactor, there's an actual change. `KeyMax` is now valid (at least in this bit of code).